src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitDao;
+import core.basesyntax.db.FruitDaoImpl;
+import core.basesyntax.service.ReportGenerator;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport() {
+        FruitDao fruitDao = new FruitDaoImpl();
+        String listFromStorage = fruitDao.getStorageQuantity().keySet().stream()
+                .map(k -> k + "," + fruitDao.getStorageQuantity().get(k) + "\n")
```

Comment:
The `getStorageQuantity()` method is called twice for each key in the stream. Consider storing the result of `fruitDao.getStorageQuantity()` in a local variable to avoid redundant method calls and improve efficiency.